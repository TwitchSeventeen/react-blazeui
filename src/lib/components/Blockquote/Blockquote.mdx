---
name: Blockquote
route: /typography/blockquote
menu: Typography
---

import { Playground, PropsTable } from 'docz'
import Blockquote from './index';

# Blockquote

Use this to style a blockquote, optionally you can attribute the quote with a citation.

## Basic usage
<Playground>
    <Blockquote>
        To quote use the following code to make it stand out from the rest of the page
    </Blockquote>
</Playground>

## Citation
Use the **cite** prop to add a footer citation to the quote.
<Playground>
    <Blockquote cite="~ Greg">
        To quote use the following code to make it stand out from the rest of the page
    </Blockquote>
</Playground>

## Properties
<PropsTable of={Blockquote} />