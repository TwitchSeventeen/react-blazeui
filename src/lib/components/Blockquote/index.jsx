import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Blockquote(props) {
  const {
    children, cite, className, dataName, ...other
  } = props;
  const classes = classNames('u-blockquote', className);
  return (
    <blockquote
      {...other}
      className={classes}
      data-name={dataName}
    >
      <div className="u-blockquote__body">
        { children }
      </div>
      { cite
          && <footer className="u-blockquote__footer">{ cite }</footer>}
    </blockquote>
  );
}

Blockquote.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Footer citation for the quote
   */
  cite: PropTypes.string,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Blockquote.defaultProps = {
  children: null,
  cite: null,
  className: null,
  dataName: 'blockquote',
};

export default Blockquote;
