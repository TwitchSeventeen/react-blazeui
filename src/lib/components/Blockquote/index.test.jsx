import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Blockquote from './index';

describe('Blockquote', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Blockquote).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Blockquote cite="~ Greg">
        lorem ipsum dolor
      </Blockquote>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly without a footer', () => {
    const { asFragment } = render(
      <Blockquote>
        lorem ipsum dolor
      </Blockquote>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
