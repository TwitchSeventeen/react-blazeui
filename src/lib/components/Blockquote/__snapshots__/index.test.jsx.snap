// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote renders correctly 1`] = `
<DocumentFragment>
  <blockquote
    class="u-blockquote"
    data-name="blockquote"
  >
    <div
      class="u-blockquote__body"
    >
      lorem ipsum dolor
    </div>
    <footer
      class="u-blockquote__footer"
    >
      ~ Greg
    </footer>
  </blockquote>
</DocumentFragment>
`;

exports[`Blockquote renders correctly without a footer 1`] = `
<DocumentFragment>
  <blockquote
    class="u-blockquote"
    data-name="blockquote"
  >
    <div
      class="u-blockquote__body"
    >
      lorem ipsum dolor
    </div>
  </blockquote>
</DocumentFragment>
`;
